src/network.rs

```diff
@@ -75,34 +75,46 @@ fn get_wait_time_exp(retrycount: i32) -> u64 {
 
 /// Process CDN response: handle retries in case of server error, or early return for client errors
 /// Successful retry data is stored into res
-fn error_handler(res: &mut Response, rb: Request) -> Result<&Response, ProjError> {
+fn error_handler<'a>(
+    res: &'a mut http::Response<ureq::Body>,
+    url: &str,
+    headers: &[(&str, &str)],
+    clt: Agent,
+) -> Result<&'a mut http::Response<ureq::Body>, ProjError> {
     let mut retries = 0;
     // Check whether something went wrong on the server, or if it's an S3 retry code
-    if SERVER_ERROR_CODES.contains(&res.status()) || RETRY_CODES.contains(&res.status()) {
+    if SERVER_ERROR_CODES.contains(&res.status().as_u16())
+        || RETRY_CODES.contains(&res.status().as_u16())
+    {
         // Start retrying: up to MAX_RETRIES
-        while (SERVER_ERROR_CODES.contains(&res.status()) || RETRY_CODES.contains(&res.status()))
+        while (SERVER_ERROR_CODES.contains(&res.status().as_u16())
+            || RETRY_CODES.contains(&res.status().as_u16()))
             && retries <= MAX_RETRIES
         {
             retries += 1;
             let wait = time::Duration::from_millis(get_wait_time_exp(retries as i32));
```

Comment:
Unrelated, but exponential back-off (a very popular approach) picks a random time between 0 and an exponentially-increasing limit.

Reply:
Ha fair. Rather than "fix" it, I've renamed it to `_quad` and updated the docstring.

Reply:
Oh wow, I didn't even notice it's `retries^2`, not `2^retries`.